src/python/pants/backend/python/rules/coverage.py

```diff
@@ -323,43 +325,50 @@ async def generate_coverage_report(
         ),
     )
 
-    report_type = coverage_subsystem.report
-    process = coverage_setup.pex.create_process(
-        pex_path=f"./{coverage_setup.pex.output_filename}",
-        # We pass `--ignore-errors` because Pants dynamically injects missing `__init__.py` files
-        # and this will cause Coverage to fail.
-        pex_args=(report_type.report_name, "--ignore-errors"),
-        input_digest=input_digest,
-        output_directories=("htmlcov",),
-        output_files=("coverage.xml",),
-        description=f"Generate Pytest {report_type.report_name} coverage report.",
-        python_setup=python_setup,
-        subprocess_encoding_environment=subprocess_encoding_environment,
-    )
-    result = await Get(ProcessResult, Process, process)
+    coverage_reports: List[CoverageReport] = []
+    for report_type in coverage_subsystem.reports:
+        process = coverage_setup.pex.create_process(
+            pex_path=f"./{coverage_setup.pex.output_filename}",
+            # We pass `--ignore-errors` because Pants dynamically injects missing `__init__.py` files
+            # and this will cause Coverage to fail.
+            pex_args=(report_type.report_name, "--ignore-errors"),
+            input_digest=input_digest,
+            output_directories=("htmlcov",),
+            output_files=("coverage.xml",),
+            description=f"Generate Pytest {report_type.report_name} coverage report.",
+            python_setup=python_setup,
+            subprocess_encoding_environment=subprocess_encoding_environment,
+        )
+        result = await Get(ProcessResult, Process, process)
```

Comment:
Instead of a `for` loop, it's much better with the engine to use `MultiGet` so that the processes can run in parallel. At a high level, something like this:

```
processes = (pex.create_process(HTML), pex.create_process(XML))
results = await MultiGet(processes)
for result in results:
   # analyze the result
```

Reply:
mypy complains about this:

<img width="1043" alt="Screenshot 2020-06-25 20 58 29" src="https://user-images.githubusercontent.com/1268088/85818863-aa2df400-b726-11ea-9d4a-8b23c26c01ee.png">

Reply:
nevermind... I figured it out..
the way MultiGet is defined (I did got to definition in VS Code) and it got me [here](https://github.com/pantsbuild/pants/blob/3c191e4958446f614b569466f18a3004ea7d298e/src/python/pants/engine/selectors.py#L223)

It is not obvious to me (a noob to pants plugin dev) how to exactly call MultiGet

Reply:
Two ways to call it: a heterogeneous list of arguments, like `await MultiGet(Get(ProcessResult, ..), Get(Snapshot, ..))` or a homogenous iterable, like `await MulitGet(Get(Snapshot, ..) for x in xs)`.

This will go in the upcoming plugin docs.